build-system/pr-check/utils.js

```diff
@@ -234,111 +243,113 @@ const timedExecOrDie = timedExecFn(execOrDie);
 const timedExecOrThrow = timedExecFn(execOrThrow);
 
 /**
- * Download output helper
- * @param {string} outputFileName
- * @param {string} outputDirs
- * @private
+ * Fetches and merges build outputs from previous jobs.
  */
-function downloadOutput_(outputFileName, outputDirs) {
+function fetchBuildOutput() {
   const loggingPrefix = getLoggingPrefix();
-  const buildOutputDownloadUrl = `${GCLOUD_STORAGE_BUCKET}/${outputFileName}`;
-  const dirsToUnzip = outputDirs.split(' ');
 
   logWithoutTimestamp(
-    `${loggingPrefix} Downloading build output from ` +
-      cyan(buildOutputDownloadUrl) +
-      '...'
+    `\n${loggingPrefix} Fetching and merging the following builds from the workspace:`
   );
-  execOrDie(`gsutil -q cp ${buildOutputDownloadUrl} ${outputFileName}`);
+  for (const containerDir of fs.readdirSync('/tmp/workspace/builds')) {
+    logWithoutTimestamp('*', cyan(containerDir));
+  }
 
-  logWithoutTimestamp(
-    `${loggingPrefix} Extracting ` + cyan(outputFileName) + '...'
-  );
-  dirsToUnzip.forEach((dir) => {
-    execOrDie(`unzip -q -o ${outputFileName} '${dir.replace('/', '/*')}'`);
-  });
-  execOrDie(`du -sh ${outputDirs}`);
+  if (isCircleciBuild()) {
+    for (const containerDir of fs.readdirSync('/tmp/workspace/builds')) {
```

Comment:
Is `/tmp/workspace/builds` stored as a CircleCI env var? Wondering if it's worth moving the fetch to a step on the CircleCI job?

Reply:
I can see the value in splitting this as a step! Let me tinker with it and see if this can be done safely in this PR

(and some background if you care for it)

Workspaces are one of the three ways that CircleCI can persist files (tl;dr is "workspace" passes files between jobs in the same workflow using a simple directory structure, "cache" passes files between different workflows using a key/value scheme, and "artifacts" are assets for consumption outside of CircleCI)

`/tmp/workspace` is just the name that I chose for the directory (it might as well be `/kitties/are/cute`)